worker/dyad-component-selector-client.js

```diff
@@ -146,6 +146,37 @@
     );
   }
 
+  function onKeyDown(e) {
+    // Ignore keystrokes if the user is typing in an input field, textarea, or editable element
+    if (
+      e.target.tagName === "INPUT" ||
+      e.target.tagName === "TEXTAREA" ||
+      e.target.isContentEditable
+    ) {
+      return;
+    }
+
+    // Forward shortcuts to parent window
+    const key = e.key.toLowerCase();
+    const hasShift = e.shiftKey;
+    const hasCtrlOrMeta = e.ctrlKey || e.metaKey;
+    if (key && hasShift && hasCtrlOrMeta) {
+      e.preventDefault();
+      window.parent.postMessage(
+        {
+          type: "dyad-shortcut-triggered",
+          key: key,
+          eventModifiers: {
+            ctrl: e.ctrlKey,
```

Comment:
Just to be safe, rather than posting a generic "dyad-shortcut-triggered" message, I would post a specific message like "dyad-select-component-shortcut" and not send the actual key (or event modifiers) to the parent window. 

See the graphite-app security comment for more context.